### PR TITLE
Deleting blog from db and getting list of all blogs

### PR DESCRIPTION
**Pull request by ArsenDriashkaba**
DeletingBlogBiID - route
GettingAllBlogs - route

### CODE REVIEW

**Comment by jakubarendac** [comment]
please extend .gitignore file with proper node config -> .DS_store should not be stored in git. It is OS related file

File: package.json [hunk 1]
```diff
@@ -5,10 +5,9 @@
   "main": "index.js",
   "type": "module",
   "scripts": {
-    "dev": "nodemon --exec node --experimental-specifier-resolution=node src/index.js | pino-pretty",
+    "dev": "nodemon --exec node --experimental-specifier-resolution=node src/index.js",
     "test": "echo \"Error: no test specified\" && exit 1",
-    "start": "node --experimental-specifier-resolution=node src/index"
-
+    "start": "nodemon --exec node --experimental-specifier-resolution=node src/index"
```
Comment (jakubarendac) [inline package.json]: what is difference between dev and start command ? If there is none please remove one of them

File: src/index.js [hunk 2]
```diff
@@ -17,7 +20,11 @@ app.get("/", (req, res) => {
   res.send("Chellou");
 });
 
-app.use('/blogs', routes.blogs)
+app.get("/blogs", services.blogs.getAllBlogs);
+
+app.delete("/:id", param("id").isLength(24), services.blogs.deleteBlogById);
```
Comment (jakubarendac) [inline src/index.js]: does this check for length make sense ? Isn't this a number ?

File: src/models/blog.js [hunk 3]
```diff
@@ -10,37 +10,34 @@ client.connect(err => {
...
-    {
-        data: Buffer,
-        contentType: String
-    }
-})
+  title: {
+    type: String,
+    required: true,
+  },
+  author: {
```
Comment (jakubarendac) [inline src/models/blog.js:8]: I don't think this is good schema for connection between blog and author. Author should be different schema and there should be a relation between these two schemas

File: src/models/blog.js [hunk 4]
```diff
@@ -10,37 +10,34 @@ client.connect(err => {
...
+    type: String,
+    required: true,
+  },
+  text: {
+    type: String,
+    required: true,
+  },
+  comments: [{ body: String, date: Date }],
```
Comment (jakubarendac) [inline src/models/blog.js:16]: Maybe we should consider adding the authors of comments to schema so we have information who added comment (if only registered users can add comments)

File: src/models/blog.js [hunk 5]
```diff
@@ -10,37 +10,34 @@ client.connect(err => {
 
 //mongodb+srv://User1:<password>@test.fnhqc.mongodb.net/myFirstDatabase?retryWrites=true&w=majority
 
-
-
-const mongoose = require('mongoose')
+// const mongoose = require('mongoose')
```
Comment (mhalajova) [inline src/models/blog.js]: remove commented line

File: src/models/blog.js [hunk 6]
```diff
@@ -10,37 +10,34 @@ client.connect(err => {
 
 //mongodb+srv://User1:<password>@test.fnhqc.mongodb.net/myFirstDatabase?retryWrites=true&w=majority
```
Comment (mhalajova) [inline src/models/blog.js]: remove a commented chunk of code.. it doesn't serve any purpose and makes the code messy

File: src/models/blog.js [hunk 7]
```diff
@@ -10,37 +10,34 @@ client.connect(err => {
...
+    data: Buffer,
+    contentType: String,
+  },
+});
+
+export const blogModel = mongoose.model("blogs", blogSchema);
 
-module.exports = mongoose.model('Blogs', blogSchema);
\ No newline at end of file
+export default blogModel;
```
Comment (mhalajova) [inline src/models/blog.js]: I would export both schema and model

```suggestion
export default { blogModel, blogSchema  as Blog };
```

File: src/services/blogs.js [hunk 8]
```diff
@@ -1,3 +1,40 @@
...
 
-export default {}
+// Function find all blogs in collection and send them as response
+const getAllBlogs = async (req, res) => {
+  try {
+    const allBlogs = await Blog.find({});
+
+    res.status(200).send(allBlogs);
+  } catch (error) {
+    res.status(500).send("My bad bro ._.");
```
Comment (mhalajova) [inline src/services/blogs.js]: Even though your comment is funny... in the production code, you should send a meaningful response message to the user.

Change the message

File: src/services/blogs.js [hunk 9]
```diff
@@ -1,3 +1,40 @@
...
+    const idOfBlog = req.params.id.toString().trim();
+
+    if (!validationResults.isEmpty()) {
+      res.status(400).send(`"${idOfBlog}" is not a valid blogID value`);
+
+      return;
+    }
+
+    //deleting blog from collection
+    await Blog.findByIdAndRemove(idOfBlog).exec();
```
Comment (mhalajova) [inline src/services/blogs.js]: Handle the use case that the blog with the given ID doesn't exist.

File: src/services/blogs.js [hunk 10]
```diff
@@ -1,3 +1,40 @@
...
+
+    res
+      .status(200)
+      .send(`You've succsesfully deleted blog with id: ${req.params.id}`);
+  } catch (error) {
+    res.status(500).send(`My bad bro ._.`);
```
Comment (mhalajova) [inline src/services/blogs.js]: Change the message

**Review by mhalajova — CHANGES_REQUESTED** [review]

File: src/services/blogs.js (hunk 8 above)
Comment (jakubarendac) [inline src/services/blogs.js]: I think we need to integrate logger and log if errors happens

**Comment by mhalajova** [comment]
Use logger. For example `req.log.info("Success")`.  Get used to logging significant events.

**Review by jakubarendac — CHANGES_REQUESTED** [review]

File: src/index.js (hunk 2 above)
Comment (ArsenDriashkaba) [reply src/index.js]: Id is generated automatically and is a string that incudes letter and number characters (example: "61dd3d63f6cc3687326c5494")
isLength validation fixed to _.isLength({min: 24, max:24})
The previous implementation wasn't working properly

File: src/models/blog.js [hunk 11]
```diff
@@ -40,4 +27,4 @@ const blogSchema = mongoose.Schema({
 
 export const blogModel = mongoose.model("blogs", blogSchema);
 
-export default blogModel;
+export default { blogModel, blogSchema };
```
Comment (mhalajova) [inline src/models/blog.js]: you can remove this line

File: src/services/blogs.js [hunk 12]
```diff
@@ -19,21 +21,31 @@ const deleteBlogById = async (req, res) => {
     const idOfBlog = req.params.id.toString().trim();
 
     if (!validationResults.isEmpty()) {
-      res.status(400).send(`"${idOfBlog}" is not a valid blogID value`);
+      req.log.info(validationResults);
```
Comment (mhalajova) [inline src/services/blogs.js]: this should be `req.log.err` or `req.log.error`

File: src/services/blogs.js [hunk 13]
```diff
@@ -19,21 +21,31 @@ const deleteBlogById = async (req, res) => {
...
 
-    res
-      .status(200)
-      .send(`You've succsesfully deleted blog with id: ${req.params.id}`);
-  } catch (error) {
-    res.status(500).send(`My bad bro ._.`);
+    if (deletedBlog == null) {
+      const blogNotFoundMsg = `Blog with id "${idOfBlog}" is not found...`;
+
+      req.log.info(blogNotFoundMsg);
```
Comment (mhalajova) [inline src/services/blogs.js]: this should be `req.log.err` or `req.log.error`

File: src/services/blogs.js [hunk 14]
```diff
@@ -19,21 +21,31 @@ const deleteBlogById = async (req, res) => {
...
-    console.log(error);
+      return;
+    }
+
+    const successMsg = `You've succsesfully deleted blog with id: ${req.params.id}`;
+
+    req.log.info(successMsg);
+    res.status(200).send(successMsg);
+  } catch (error) {
+    req.log.info(error);
```
Comment (mhalajova) [inline src/services/blogs.js]: this should be `req.log.err` or `req.log.error`

**Comment by mhalajova** [comment]
👍

**Review by jakubarendac — APPROVED** [review]

**Review by mhalajova — CHANGES_REQUESTED** [review]

**Pull request merged**